packages/snaps-ui/src/builder.test.ts

```diff
@@ -242,3 +250,36 @@ describe('text', () => {
     );
   });
 });
+
+describe('image', () => {
+  const MOCK_SVG = '<svg />';
+  it('creates an image component', () => {
+    expect(image({ value: MOCK_SVG })).toStrictEqual({
+      type: NodeType.Image,
+      value: MOCK_SVG,
+    });
+  });
+
+  it('creates an image component using the shorthand form', () => {
+    expect(image(MOCK_SVG)).toStrictEqual({
+      type: NodeType.Image,
+      value: MOCK_SVG,
+    });
+  });
+
+  it('validates the args', () => {
+    expect(() => image({ value: 'foo' })).toThrow(
+      'Invalid image component: At path: value -- Expected a value of type `string`, but received: `"foo"`.',
```

Comment:
This error is not very helpful. With a custom refiner you could return a better error message.

Reply:
good idea - fixed!